src/arena.rs

```diff
@@ -51,6 +51,12 @@ impl Index {
 
         Self { generation, slot }
     }
+
+    /// Convert this `Index` to its u32 slot index, discarding the `Generation`
+    /// info.
```

Comment:
The wording on this method is kind of tricky given that we don't export all of this module's items from the crate. This might also be the first time "slot" is appearing in the public API, so defining it might be useful.

What do you think of a comment like:

```
Convert this `Index` into a slot, discarding its generation. Slots describe a
location in an [`Arena`] and are reused when entries are removed.
```

Reply:
That sounds good to me! I'll slap it in there. Wasn't quite sure how to approach that definition so if that works for you, I'm good with it.

Reply:
It seems alright to me as a starting point. Maybe we'll add a section to `lib.rs` eventually to be an overview of the terms and types in the crate.

Reply:
Yeah, that seems like the right approach. Maybe before 0.4?